packages/engineer/src/feature/dev-server.dev-server.env.ts

```diff
@@ -90,14 +90,18 @@ devServerFeature.setup(
             //Node environment manager, need to add self to the topology, I thing starting the server and the NEM should happen in the setup and not in the run
             // So potential dependencies can rely on them in the topology
             application.setNodeEnvManager(
-                new NodeEnvironmentsManager(socketServer, {
-                    configurations,
-                    features,
-                    defaultRuntimeOptions,
-                    port: actualPort,
-                    inspect,
-                    overrideConfig,
-                })
+                new NodeEnvironmentsManager(
+                    socketServer,
+                    {
+                        configurations,
+                        features,
+                        defaultRuntimeOptions,
+                        port: actualPort,
+                        inspect,
+                        overrideConfig,
+                    },
+                    { ...socketServerOptions, ...configServerOptions }
```

Comment:
shouldn't the destructuring be the other way around? (also in code above)
also, it can be calculated once.

Reply:
Don't we want the provided socket options to override the default ones (currently `undefined`)?

Reply:
`{...(engine.config.js), ...(feature.config.js)}`

Reply:
so that specific config files will be able to override repo-wide engine config...